### ISSUE TITLE
request-not-cached error

### ISSUE THREAD

**Issue by diachedelic** [GoogleChrome/workbox#1000]
**Library Affected**:
workbox-sw

**Browser & Platform**:
workbox-sw@2.1.1
Google Chrome 62.0.3202.89 (Official Build) (64-bit)

**Issue or Feature Request Description**:
Upon loading my webapp with no installed service workers, I get this error in the console:

```
service-worker.js:279 Uncaught (in promise) WorkboxError: An error was thrown by workbox with error code: ;'request-not-cached' with extras: '{"url":"https://beta-toolkit2.crosscountryapp.com/index.html","error":{}}'
    at https://beta-toolkit2.crosscountryapp.com/service-worker.js:281:1358
    at Generator.throw (<anonymous>)
    at step (https://beta-toolkit2.crosscountryapp.com/service-worker.js:204:30)
    at https://beta-toolkit2.crosscountryapp.com/service-worker.js:217:13
    at <anonymous>
```

Happens on localhost too, and I've tried clearing the storage from `DevTools -> Application -> Clear Storage`.  I rolled back my version of Chrome from Beta to Stable but the issue has persisted.  I don't know why it's happening now - it was working quite recently.

My essential [service-worker.js](https://beta-toolkit2.crosscountryapp.com/service-worker.js) after build:

```javascript
import WorkboxSW from 'workbox-sw'
const workboxSW = new WorkboxSW()

workboxSW.precache([
  {
    "url": "/index.html",
    "revision": "831d10db02fa01fde6b0a2e1ba3de1ef"
  },
  {
    "url": "/static/css/app.c2c8b350bf27b9075d0b404e09ac192d.css",
    "revision": "a938eb302a496ccd9fc20d9a02cfa225"
  },
  ...
])
```

Public URL: https://beta-toolkit2.crosscountryapp.com/?authToken=0bvyir9d8pis57jzf06punttg

**Comment by diachedelic** [comment]
There is also an earlier error that never makes it to the console, but is logged in `DevTools -> Application -> Service Workers -> Errors`

```
(unknown)#1877: ServiceWorker failed to install: Service worker failed to handle event  (event.waitUntil Promise rejected)
```

**Comment by diachedelic** [comment]
Woo hoo, thousandth issue!

**Comment by jeffposnick** [comment]
Hello @diachedelic—I just tried  https://beta-toolkit2.crosscountryapp.com/?authToken=0bvyir9d8pis57jzf06punttg in Chrome Stable, Chrome Canary, and Firefox Nightly, and the service worker successfully installed and took control of the page in all three.

Have you changed anything in the past ~13 hours since reporting this?

I noticed that you're serving your service worker with HTTP caching enabled, which I'd recommend *not* doing, as you'll continue getting your previously cached service worker for [up to 24 hours](https://stackoverflow.com/questions/38843970/service-worker-javascript-update-frequency-every-24-hours/38854905#38854905). That could make it hard to debug issues that you've resolved locally and deployed a fix for, but don't get cleared up for a day:

<img width="513" alt="screen shot 2017-11-07 at 11 12 11 am" src="https://user-images.githubusercontent.com/1749548/32503967-a74219a8-c3ac-11e7-86d9-45efa32aafce.png">

If you can reproduce it in general still, can you also reproduce in Chrome when starting in an Incognito window?

**Comment by diachedelic** [comment]
Ah!  Using an incognito window fixes the issue, however it is still occurring without.

Perhaps my Chrome is in some sort of inconsistent state?  I did reinstall it earlier and that did not fix the issue.

Thanks for the tip on caching, I'll fix that up - but for now, I'm debugging on localhost which isn't caching anything.

**Comment by diachedelic** [comment]
And no, i didn't change anything to do with the service worker since reporting.

**Comment by diachedelic** [comment]
One thing I forgot to mention - I vaguely remember the issue started occurring after trying to cache some video, which was about 12MB (using the workbox-range-requests module?) - I think I got a quote exceeded error, and since then other things have refused to cache.

**Comment by diachedelic** [comment]
Is it expected behaviour that running `DevTools -> Application -> Clear Storage -> Clear Site Data` does not change the ring graph above?

Before and after:
![image](https://user-images.githubusercontent.com/310464/32524630-ce18b730-c474-11e7-939a-e583763ce3a9.png)

**Comment by diachedelic** [comment]
Solved - I only had 3.5GB disk space left on my machine, so I freed up 15GB and everything works now.  I suppose the bug here is that the underlying DOM Quote Exceeded error was not visible.

**Comment by jeffposnick** [comment]
That error in v2 comes from https://github.com/GoogleChrome/workbox/blob/master/packages/workbox-precaching/src/lib/controllers/base-cache-manager.js#L155-L171

In v3, the equivalent code is https://github.com/GoogleChrome/workbox/blob/v3/packages/workbox-precaching/controllers/PrecacheController.mjs#L174-L177

I believe that means that in v3, you're just going to see the underlying exception directly, since we don't catch and re-throw anymore. (The original exception **should** have been logged in v2, but I've noticed that objects like `err` in that example don't always persist in their "live" form in the console's logs.)
